osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -276,7 +276,11 @@ private void nudgeSelection(int amount)
                 var timingPoint = EditorBeatmap.ControlPointInfo.TimingPointAt(selected.First().StartTime);
                 double adjustment = timingPoint.BeatLength / EditorBeatmap.BeatDivisor * amount;
 
-                EditorBeatmap.PerformOnSelection(h => h.StartTime += adjustment);
+                EditorBeatmap.PerformOnSelection(h =>
+                {
+                    h.StartTime += adjustment;
+                    EditorBeatmap.Update(h);
```

Comment:
What are all of these `Update()` calls doing? Seems like this is handled via `EndChange()` from `PerformOnSelection()` anyway?

Reply:
It's only handled if a modification is detected by `EditorBeatmap`. Check the flow of `batchPendingUpdates`, I'm pretty sure you'll find that it's required. Also see https://github.com/ppy/osu/pull/12916#discussion_r637540972 which I explain this.

Reply:
Ah, for some reason I thought I saw an add to `batchPendingUpdates` inside there and thought this was duplicating it.